src/app/archive/snapp_tables.sql

```diff
@@ -0,0 +1,205 @@
+/* snapp_tables.sql -- support tables for Snapp commands */
+
+/* Several of the tables below support the following convention, related
+   to NULL values.
+
+   In OCaml, some Snapp-related types use the constructors Check, which takes a value,
+   and Ignore, which is nullary. In columns following the convention, a NULL means Ignore, while
+   non-NULL means Check.
+
+   Similarly, in OCaml, there are the constructors Set, which takes a value, and
+   Keep, which is nullary. NULL means Keep, and non-NULL means Set.
+
+   The tables that follow this convention have a comment "NULL convention".
+*/
+
+/* the string representation of an algebraic field */
+CREATE TABLE snapp_state_data
+( id                       serial           PRIMARY KEY
+, field                    text             NOT NULL UNIQUE
+);
+
+/* Variable-width arrays of algebraic fields, given as
+   id's from snapp_state_data
+
+   Postgresql does not allow enforcing that these are
+   foreign keys.
+*/
+CREATE TABLE snapp_state_array
+( id                       serial  PRIMARY KEY
+, items                    int[]   NOT NULL
+);
+
+/* NULL convention */
+CREATE TABLE snapp_states
+( id                       serial           PRIMARY KEY
+, element_0_id             int              REFERENCES snapp_state_data(id)
```

Comment:
Mentioned in chat - but would be good to use an array here so that state size can be reconfigured easily.

Reply:
Done.